// @aws-sdk/client-appstream/dist-es/commands/AssociateApplicationFleetCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AssociateApplicationFleetRequest, AssociateApplicationFleetResult } from "../models/models_0";
import { deserializeAws_json1_1AssociateApplicationFleetCommand, serializeAws_json1_1AssociateApplicationFleetCommand, } from "../protocols/Aws_json1_1";
var AssociateApplicationFleetCommand = (function (_super) {
    __extends(AssociateApplicationFleetCommand, _super);
    function AssociateApplicationFleetCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AssociateApplicationFleetCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AppStreamClient";
        var commandName = "AssociateApplicationFleetCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AssociateApplicationFleetRequest.filterSensitiveLog,
            outputFilterSensitiveLog: AssociateApplicationFleetResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AssociateApplicationFleetCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1AssociateApplicationFleetCommand(input, context);
    };
    AssociateApplicationFleetCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1AssociateApplicationFleetCommand(output, context);
    };
    return AssociateApplicationFleetCommand;
}($Command));
export { AssociateApplicationFleetCommand };
