// @aws-sdk/client-appstream/dist-es/commands/CreateUpdatedImageCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateUpdatedImageRequest, CreateUpdatedImageResult } from "../models/models_0";
import { deserializeAws_json1_1CreateUpdatedImageCommand, serializeAws_json1_1CreateUpdatedImageCommand, } from "../protocols/Aws_json1_1";
var CreateUpdatedImageCommand = (function (_super) {
    __extends(CreateUpdatedImageCommand, _super);
    function CreateUpdatedImageCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateUpdatedImageCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AppStreamClient";
        var commandName = "CreateUpdatedImageCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateUpdatedImageRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateUpdatedImageResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateUpdatedImageCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CreateUpdatedImageCommand(input, context);
    };
    CreateUpdatedImageCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CreateUpdatedImageCommand(output, context);
    };
    return CreateUpdatedImageCommand;
}($Command));
export { CreateUpdatedImageCommand };
