// @aws-sdk/client-appstream/dist-es/commands/CreateImageBuilderStreamingURLCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateImageBuilderStreamingURLRequest, CreateImageBuilderStreamingURLResult } from "../models/models_0";
import { deserializeAws_json1_1CreateImageBuilderStreamingURLCommand, serializeAws_json1_1CreateImageBuilderStreamingURLCommand, } from "../protocols/Aws_json1_1";
var CreateImageBuilderStreamingURLCommand = (function (_super) {
    __extends(CreateImageBuilderStreamingURLCommand, _super);
    function CreateImageBuilderStreamingURLCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateImageBuilderStreamingURLCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AppStreamClient";
        var commandName = "CreateImageBuilderStreamingURLCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateImageBuilderStreamingURLRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateImageBuilderStreamingURLResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateImageBuilderStreamingURLCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CreateImageBuilderStreamingURLCommand(input, context);
    };
    CreateImageBuilderStreamingURLCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CreateImageBuilderStreamingURLCommand(output, context);
    };
    return CreateImageBuilderStreamingURLCommand;
}($Command));
export { CreateImageBuilderStreamingURLCommand };
