// @aws-sdk/client-appstream/dist-es/commands/DeleteImagePermissionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteImagePermissionsRequest, DeleteImagePermissionsResult } from "../models/models_0";
import { deserializeAws_json1_1DeleteImagePermissionsCommand, serializeAws_json1_1DeleteImagePermissionsCommand, } from "../protocols/Aws_json1_1";
var DeleteImagePermissionsCommand = (function (_super) {
    __extends(DeleteImagePermissionsCommand, _super);
    function DeleteImagePermissionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteImagePermissionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AppStreamClient";
        var commandName = "DeleteImagePermissionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteImagePermissionsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteImagePermissionsResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteImagePermissionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DeleteImagePermissionsCommand(input, context);
    };
    DeleteImagePermissionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DeleteImagePermissionsCommand(output, context);
    };
    return DeleteImagePermissionsCommand;
}($Command));
export { DeleteImagePermissionsCommand };
