// @aws-sdk/client-appstream/dist-es/commands/DeleteImageBuilderCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteImageBuilderRequest, DeleteImageBuilderResult } from "../models/models_0";
import { deserializeAws_json1_1DeleteImageBuilderCommand, serializeAws_json1_1DeleteImageBuilderCommand, } from "../protocols/Aws_json1_1";
var DeleteImageBuilderCommand = (function (_super) {
    __extends(DeleteImageBuilderCommand, _super);
    function DeleteImageBuilderCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteImageBuilderCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AppStreamClient";
        var commandName = "DeleteImageBuilderCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteImageBuilderRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteImageBuilderResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteImageBuilderCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DeleteImageBuilderCommand(input, context);
    };
    DeleteImageBuilderCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DeleteImageBuilderCommand(output, context);
    };
    return DeleteImageBuilderCommand;
}($Command));
export { DeleteImageBuilderCommand };
