// @aws-sdk/client-appstream/dist-es/commands/DeleteDirectoryConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteDirectoryConfigRequest, DeleteDirectoryConfigResult } from "../models/models_0";
import { deserializeAws_json1_1DeleteDirectoryConfigCommand, serializeAws_json1_1DeleteDirectoryConfigCommand, } from "../protocols/Aws_json1_1";
var DeleteDirectoryConfigCommand = (function (_super) {
    __extends(DeleteDirectoryConfigCommand, _super);
    function DeleteDirectoryConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteDirectoryConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AppStreamClient";
        var commandName = "DeleteDirectoryConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteDirectoryConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteDirectoryConfigResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteDirectoryConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DeleteDirectoryConfigCommand(input, context);
    };
    DeleteDirectoryConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DeleteDirectoryConfigCommand(output, context);
    };
    return DeleteDirectoryConfigCommand;
}($Command));
export { DeleteDirectoryConfigCommand };
