// @aws-sdk/client-appstream/dist-es/commands/DescribeUsersCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeUsersRequest, DescribeUsersResult } from "../models/models_0";
import { deserializeAws_json1_1DescribeUsersCommand, serializeAws_json1_1DescribeUsersCommand, } from "../protocols/Aws_json1_1";
var DescribeUsersCommand = (function (_super) {
    __extends(DescribeUsersCommand, _super);
    function DescribeUsersCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeUsersCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AppStreamClient";
        var commandName = "DescribeUsersCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeUsersRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeUsersResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeUsersCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeUsersCommand(input, context);
    };
    DescribeUsersCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeUsersCommand(output, context);
    };
    return DescribeUsersCommand;
}($Command));
export { DescribeUsersCommand };
