// @aws-sdk/client-appstream/dist-es/commands/DescribeUsageReportSubscriptionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeUsageReportSubscriptionsRequest, DescribeUsageReportSubscriptionsResult } from "../models/models_0";
import { deserializeAws_json1_1DescribeUsageReportSubscriptionsCommand, serializeAws_json1_1DescribeUsageReportSubscriptionsCommand, } from "../protocols/Aws_json1_1";
var DescribeUsageReportSubscriptionsCommand = (function (_super) {
    __extends(DescribeUsageReportSubscriptionsCommand, _super);
    function DescribeUsageReportSubscriptionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeUsageReportSubscriptionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AppStreamClient";
        var commandName = "DescribeUsageReportSubscriptionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeUsageReportSubscriptionsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeUsageReportSubscriptionsResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeUsageReportSubscriptionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeUsageReportSubscriptionsCommand(input, context);
    };
    DescribeUsageReportSubscriptionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeUsageReportSubscriptionsCommand(output, context);
    };
    return DescribeUsageReportSubscriptionsCommand;
}($Command));
export { DescribeUsageReportSubscriptionsCommand };
