// @aws-sdk/client-appstream/dist-es/commands/DescribeStacksCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeStacksRequest, DescribeStacksResult } from "../models/models_0";
import { deserializeAws_json1_1DescribeStacksCommand, serializeAws_json1_1DescribeStacksCommand, } from "../protocols/Aws_json1_1";
var DescribeStacksCommand = (function (_super) {
    __extends(DescribeStacksCommand, _super);
    function DescribeStacksCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeStacksCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AppStreamClient";
        var commandName = "DescribeStacksCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeStacksRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeStacksResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeStacksCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeStacksCommand(input, context);
    };
    DescribeStacksCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeStacksCommand(output, context);
    };
    return DescribeStacksCommand;
}($Command));
export { DescribeStacksCommand };
