// @aws-sdk/client-appstream/dist-es/commands/DisassociateApplicationFleetCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DisassociateApplicationFleetRequest, DisassociateApplicationFleetResult } from "../models/models_0";
import { deserializeAws_json1_1DisassociateApplicationFleetCommand, serializeAws_json1_1DisassociateApplicationFleetCommand, } from "../protocols/Aws_json1_1";
var DisassociateApplicationFleetCommand = (function (_super) {
    __extends(DisassociateApplicationFleetCommand, _super);
    function DisassociateApplicationFleetCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DisassociateApplicationFleetCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AppStreamClient";
        var commandName = "DisassociateApplicationFleetCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DisassociateApplicationFleetRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DisassociateApplicationFleetResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DisassociateApplicationFleetCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DisassociateApplicationFleetCommand(input, context);
    };
    DisassociateApplicationFleetCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DisassociateApplicationFleetCommand(output, context);
    };
    return DisassociateApplicationFleetCommand;
}($Command));
export { DisassociateApplicationFleetCommand };
