// @aws-sdk/client-appstream/dist-es/commands/EnableUserCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { EnableUserRequest, EnableUserResult } from "../models/models_0";
import { deserializeAws_json1_1EnableUserCommand, serializeAws_json1_1EnableUserCommand, } from "../protocols/Aws_json1_1";
var EnableUserCommand = (function (_super) {
    __extends(EnableUserCommand, _super);
    function EnableUserCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    EnableUserCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AppStreamClient";
        var commandName = "EnableUserCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: EnableUserRequest.filterSensitiveLog,
            outputFilterSensitiveLog: EnableUserResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    EnableUserCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1EnableUserCommand(input, context);
    };
    EnableUserCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1EnableUserCommand(output, context);
    };
    return EnableUserCommand;
}($Command));
export { EnableUserCommand };
