// @aws-sdk/client-appstream/dist-es/commands/ExpireSessionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ExpireSessionRequest, ExpireSessionResult } from "../models/models_0";
import { deserializeAws_json1_1ExpireSessionCommand, serializeAws_json1_1ExpireSessionCommand, } from "../protocols/Aws_json1_1";
var ExpireSessionCommand = (function (_super) {
    __extends(ExpireSessionCommand, _super);
    function ExpireSessionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ExpireSessionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AppStreamClient";
        var commandName = "ExpireSessionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ExpireSessionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ExpireSessionResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ExpireSessionCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ExpireSessionCommand(input, context);
    };
    ExpireSessionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ExpireSessionCommand(output, context);
    };
    return ExpireSessionCommand;
}($Command));
export { ExpireSessionCommand };
