// @aws-sdk/client-appstream/dist-es/commands/StartFleetCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartFleetRequest, StartFleetResult } from "../models/models_0";
import { deserializeAws_json1_1StartFleetCommand, serializeAws_json1_1StartFleetCommand, } from "../protocols/Aws_json1_1";
var StartFleetCommand = (function (_super) {
    __extends(StartFleetCommand, _super);
    function StartFleetCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StartFleetCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AppStreamClient";
        var commandName = "StartFleetCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StartFleetRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StartFleetResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StartFleetCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1StartFleetCommand(input, context);
    };
    StartFleetCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1StartFleetCommand(output, context);
    };
    return StartFleetCommand;
}($Command));
export { StartFleetCommand };
