// @aws-sdk/client-appstream/dist-es/commands/StopFleetCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StopFleetRequest, StopFleetResult } from "../models/models_0";
import { deserializeAws_json1_1StopFleetCommand, serializeAws_json1_1StopFleetCommand } from "../protocols/Aws_json1_1";
var StopFleetCommand = (function (_super) {
    __extends(StopFleetCommand, _super);
    function StopFleetCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StopFleetCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AppStreamClient";
        var commandName = "StopFleetCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StopFleetRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StopFleetResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StopFleetCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1StopFleetCommand(input, context);
    };
    StopFleetCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1StopFleetCommand(output, context);
    };
    return StopFleetCommand;
}($Command));
export { StopFleetCommand };
