// @aws-sdk/client-appstream/dist-es/commands/StopImageBuilderCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StopImageBuilderRequest, StopImageBuilderResult } from "../models/models_0";
import { deserializeAws_json1_1StopImageBuilderCommand, serializeAws_json1_1StopImageBuilderCommand, } from "../protocols/Aws_json1_1";
var StopImageBuilderCommand = (function (_super) {
    __extends(StopImageBuilderCommand, _super);
    function StopImageBuilderCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StopImageBuilderCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AppStreamClient";
        var commandName = "StopImageBuilderCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StopImageBuilderRequest.filterSensitiveLog,
            outputFilterSensitiveLog: StopImageBuilderResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StopImageBuilderCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1StopImageBuilderCommand(input, context);
    };
    StopImageBuilderCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1StopImageBuilderCommand(output, context);
    };
    return StopImageBuilderCommand;
}($Command));
export { StopImageBuilderCommand };
