// @aws-sdk/client-appstream/dist-es/commands/UpdateDirectoryConfigCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateDirectoryConfigRequest, UpdateDirectoryConfigResult } from "../models/models_0";
import { deserializeAws_json1_1UpdateDirectoryConfigCommand, serializeAws_json1_1UpdateDirectoryConfigCommand, } from "../protocols/Aws_json1_1";
var UpdateDirectoryConfigCommand = (function (_super) {
    __extends(UpdateDirectoryConfigCommand, _super);
    function UpdateDirectoryConfigCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateDirectoryConfigCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AppStreamClient";
        var commandName = "UpdateDirectoryConfigCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateDirectoryConfigRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateDirectoryConfigResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateDirectoryConfigCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdateDirectoryConfigCommand(input, context);
    };
    UpdateDirectoryConfigCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdateDirectoryConfigCommand(output, context);
    };
    return UpdateDirectoryConfigCommand;
}($Command));
export { UpdateDirectoryConfigCommand };
