// @aws-sdk/client-appstream/dist-es/commands/UpdateFleetCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateFleetRequest, UpdateFleetResult } from "../models/models_0";
import { deserializeAws_json1_1UpdateFleetCommand, serializeAws_json1_1UpdateFleetCommand, } from "../protocols/Aws_json1_1";
var UpdateFleetCommand = (function (_super) {
    __extends(UpdateFleetCommand, _super);
    function UpdateFleetCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateFleetCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AppStreamClient";
        var commandName = "UpdateFleetCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateFleetRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdateFleetResult.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateFleetCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdateFleetCommand(input, context);
    };
    UpdateFleetCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdateFleetCommand(output, context);
    };
    return UpdateFleetCommand;
}($Command));
export { UpdateFleetCommand };
